app/controllers/api/v1/targets_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Api
+  module V1
+    class TargetsController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_target, only: [:show]
+
+      def index
+        @targets = Target.all
+      end
+
+      def show; end
+
+      def create
+        new_target
+        @target.save ? render_create_success : render_create_error
+      end
+
+      private
+
+      def set_target
+        @target = Target.find(target_params[:title])
+      end
+
+      def target_params
+        params.require(:target).permit(:id, :user_id, :topic_id, :title, :radius,
+                                       :latitude, :longitude)
+      end
+
+      def new_target
+        @target = Target.new(target_params.except(:id))
```

Comment:
You can put this right on the action method and use `current_user.targets.create!(target_params)` as suggested above